src/bindings/python/PyLut1DTransform.cpp

```diff
@@ -0,0 +1,118 @@
+                         bool isRawHalfs,
+                         BitDepth fileOutputBitDepth,
+                         Lut1DHueAdjust hueAdjust,
+                         Interpolation interpolation,
+                         TransformDirection dir)
+            {
+                Lut1DTransformRcPtr p = Lut1DTransform::Create(length, isHalfDomain);
+                p->setOutputRawHalfs(isRawHalfs);
+                p->setFileOutputBitDepth(fileOutputBitDepth);
+                p->setHueAdjust(hueAdjust);
+                p->setInterpolation(interpolation);
+                p->setDirection(dir);
+                return p;
+            }),
+             "length"_a = DEFAULT->getLength(),
+             "isHalfDomain"_a = DEFAULT->getInputHalfDomain(),
+             "isRawHalfs"_a = DEFAULT->getOutputRawHalfs(),
+             "fileOutputBitDepth"_a = DEFAULT->getFileOutputBitDepth(),
+             "hueAdjust"_a = DEFAULT->getHueAdjust(),
+             "interpolation"_a = DEFAULT->getInterpolation(),
+             "dir"_a = DEFAULT->getDirection())
+
+        .def("getFileOutputBitDepth", &Lut1DTransform::getFileOutputBitDepth)
+        .def("setFileOutputBitDepth", &Lut1DTransform::setFileOutputBitDepth, "bitDepth"_a)
+        .def("getFormatMetadata", 
+             (FormatMetadata & (Lut1DTransform::*)()) &Lut1DTransform::getFormatMetadata,
+             py::return_value_policy::reference_internal)
+        .def("getFormatMetadata", 
+             (const FormatMetadata & (Lut1DTransform::*)() const) 
+             &Lut1DTransform::getFormatMetadata,
+             py::return_value_policy::reference_internal)
+        .def("equals", &Lut1DTransform::equals, "other"_a)
+        .def("getLength", &Lut1DTransform::getLength)
+        .def("setLength", &Lut1DTransform::setLength, "length"_a)
+        .def("getValue", [](Lut1DTransformRcPtr & self, unsigned long index) 
+            {
+                float r, g, b;
+                self->getValue(index, r, g, b);
+                return py::make_tuple(r, g, b);
+            }, 
+            "index"_a)
+        .def("setValue", &Lut1DTransform::setValue, "index"_a, "r"_a, "g"_a, "b"_a)
+        .def("setData", [](Lut1DTransformRcPtr & self, py::buffer & data) 
+            {
+                py::buffer_info info = data.request();
+                checkBufferType(info, py::dtype("float32"));
+                checkBufferDivisible(info, 3);
+
+                py::gil_scoped_release release;
+
+                unsigned long size = (info.size >= 0 ? static_cast<unsigned long>(info.size) : 0);
+                float * values = static_cast<float *>(info.ptr);
+
+                self->setLength(size / 3);
+
+                for (size_t i = 0; i < size; i += 3)
+                {
+                    self->setValue(i / 3, values[i], values[i+1], values[i+2]);
+                }
+            },
+             "data"_a)
+        .def("getData", [](Lut1DTransformRcPtr & self) 
```

Comment:
No problem. I tried to release the GIL anywhere the bindings are doing large array iteration or copying. It may not be needed in practice, but if there is ever a case where someone wants to utilize Python threading to operate on a Lut*Transform in parallel, it is supported. I have very much appreciated @lgritz use of `py::gil_scoped_release` in the OIIO Python bindings.

Reply:
And as an explanation of why I chose to add this method, which doesn't exist in the C++ API, it was just because Python iteration is such a performance bottleneck, so `getValue` on its own would be more problematic than good for anything other than querying a specific LUT entry.

Reply:
This will be very useful with NumPy, so it's a great add.